docs/formatter.md

```diff
@@ -0,0 +1,648 @@
+
+## Configuration
+
+The Ruff Formatter exposes a small set of configuration options, some of which are also supported
+by Black (like line width), some of which are unique to Ruff (like quote and indentation style).
+
+For example, to configure the formatter to use double single quotes, a line width of 100, and
+tab indentation, add the following to your `pyproject.toml`:
+
+```toml
+[tool.ruff.format]
+line-length = 100
+quote-style = "single"
+indent-style = "tab"
+```
+
+For more on configuring Ruff via `pyproject.toml`, see [_Configuring Ruff_](configuration.md).
+
+Given the focus on Black compatibility (and unlike formatters like [YAPF](https://github.com/google/yapf)),
+Ruff does not currently expose any configuration options to modify core formatting behavior outside
+of these trivia-related settings.
+
+## Format suppression
+
+Like Black, Ruff supports `# fmt: on`, `# fmt: off`, and `# fmt: skip` pragma comments, which can
+be used to temporarily disable formatting for a given code block.
+
+`# fmt: on` and `# fmt: off` comments are enforced at the statement level:
+
+```python
+# fmt: off
+not_formatted=3
+also_not_formatted=4
+# fmt:on
+```
+
+As such, adding `# fmt: on` and `# fmt: off` comments within expressions will have no effect. In
+the following example, both list entries will be formatted, despite the `# fmt: off`:
+
+```python
+[
+    # fmt: off
+    '1',
+    # fmt: on
+    '2',
+]
+```
+
+Instead, apply the `# fmt: off` comment to the entire statement:
+
+```python
+# fmt: off
+[
+    '1',
+    '2',
+]
+```
+
+`# fmt: skip` comments suppress formatting for a preceding statement, case header, decorator,
+function definition, or class definition:
+
+```python
+if True:
+    pass
+elif False: # fmt: skip
+    pass
+
+@Test
+@Test2 # fmt: off
+def test(): ...
+
+a = [1, 2, 3, 4, 5] # fmt: off
+
+def test(a, b, c, d, e, f) -> int: # fmt: skip
+    pass
+```
+
+Like Black, Ruff will _also_ recognize [YAPF](https://github.com/google/yapf)'s `# yapf: disable` and `# yapf: enable` pragma
+comments, which are treated equivalently to `# fmt: off` and `# fmt: on`, respectively.
+
+## Conflicting lint rules
+
+The Ruff Formatter is designed to be used alongside the Ruff Linter. However, the linter includes
+some rules that, when enabled, can cause conflicts with the formatter, leading to unexpected
+behavior.
+
+When using Ruff as a formatter, we recommend disabling the following rules:
+
+- [`line-too-long`](rules/line-too-long.md)
+- [`single-line-implicit-string-concatenation`](rules/single-line-implicit-string-concatenation.md)
+- [`missing-trailing-comma`](rules/missing-trailing-comma.md)
+- [`prohibited-trailing-comma`](rules/prohibited-trailing-comma.md)
+- [`bad-quotes-inline-string`](rules/bad-quotes-inline-string.md)
+- [`bad-quotes-multiline-string`](rules/bad-quotes-multiline-string.md)
+- [`bad-quotes-docstring`](rules/bad-quotes-docstring.md)
+- [`avoidable-escaped-quote`](rules/avoidable-escaped-quote.md)
```

Comment:
These aren't _necessarily_ conflicting with the formatter. The quote rules are probably consistent with it, assuming you have the same quote settings. And the trailing comma rules won't _conflict_ with the formatter, they'll just end up changing the resulting formatting.